go.mod

```diff
@@ -12,23 +10,44 @@ require (
 	github.com/posener/complete/v2 v2.1.0
 	github.com/sethvargo/go-envconfig v1.0.1
 	github.com/sethvargo/go-retry v0.2.4
-	golang.org/x/oauth2 v0.19.0
-	golang.org/x/sync v0.7.0
-	golang.org/x/text v0.14.0
-	google.golang.org/grpc v1.63.2
-	google.golang.org/protobuf v1.34.0
+	go.opentelemetry.io/contrib/detectors/gcp v1.32.0
```

Comment:
This is a LOT of new dependencies. We try really hard to keep `pkg`'s dependencies small because it's used in so many places. I'm wondering if this should be its own package.

Reply:
Thanks for taking a look!

>  I'm wondering if this should be its own package.
Do you mean move it to a separate repo?  Thing is this acts as a shared library much like [logging](https://github.com/abcxyz/pkg/tree/main/logging) so I would hope it lives here as well.

Reply:
`logging` only has 3 external dependencies, vs 15 here.

As I understand, anybody who is copying `pkg` will have do download all of these as well. Note that they will not end up in the binary if not used.

Reply:
Yes to what @pdewilde said, and it has a high chance of conflicting (diamond dependency problem) given the number of floating otel libraries out there.

Reply:
If so, it seems like our options are to provide either (1) a separate abcxyz repo or (2) a template.

Is it normal to support an abcxyz template for Go code though? From what I understand, it caters more to TF, yaml, and other config languages.

Reply:
Addressed in https://github.com/abcxyz/pkg/pull/357#issuecomment-2542084312